var/spack/repos/builtin/packages/llvm-amdgpu/package.py

```diff
@@ -163,6 +165,7 @@ class LlvmAmdgpu(CMakePackage):
     # as per 5.2.0 llvm code. It used to be llvm/bin/../lib/libdevice.
     # Below patch is to look in the old path.
     patch("adjust-openmp-bitcode-directory-for-llvm-link.patch", when="@5.2.0:")
+    patch("patch-llvm-5.5.0.patch", when="@5.5.0")
```

Comment:
This should have a more descriptive filename and a comment explaining why it is needed.

Reply:
basically, with llvm-amdgpu@5.5.0, it seems to support the code-object-version 5 but for some reason the changes in the hsa-rocr-dev for 5.5.0 in the file  https://github.com/RadeonOpenCompute/ROCR-Runtime/blob/master/src/image/blit_src/CMakeLists.txt seems to break for spack build when building hsa-rocr-dev.  Rightnow, this workaround is put for this reason. Still checking if there is anything that can be done.

Reply:
basically this line inside https://github.com/RadeonOpenCompute/ROCR-Runtime/blob/master/src/image/blit_src/CMakeLists.txt 

 separate_arguments(CLANG_ARG_LIST UNIX_COMMAND
    "-O2 -x cl -Xclang -finclude-default-header -cl-denorms-are-zero -cl-std=CL2.0
    -target amdgcn-amd-amdhsa -mcpu=${TARGET_ID} **-mcode-object-version=4**
    -o ${OUTPUT_FILE} ${INPUT_FILE}")